pennylane_lightning/core/src/simulators/lightning_kokkos/gates/tests/Test_StateVectorKokkos_Generator.cpp

```diff
@@ -412,3 +434,340 @@ TEMPLATE_TEST_CASE("StateVectorKokkos::applyGeneratorDoubleExcitationPlus",
+    const std::size_t control_wire = GENERATE(0, 1, 2);
+    const std::size_t wire = GENERATE(0, 1, 2);
+    SECTION("CRX") {
+        StateVectorKokkos<TestType> kokkos_gntr_sv_cr{ini_st.data(),
+                                                      ini_st.size()};
+        StateVectorKokkos<TestType> kokkos_gntr_sv_r{ini_st.data(),
+                                                     ini_st.size()};
+
+        if (control_wire == wire) {
+            PL_REQUIRE_THROWS_MATCHES(
+                kokkos_gntr_sv_r.applyGenerator("RX", {control_wire}, {true},
+                                                {wire}, inverse),
+                LightningException,
+                "`controlled_wires` and `target wires` must be disjoint.");
+        } else {
+            kokkos_gntr_sv_cr.applyGenerator("CRX", {control_wire, wire},
+                                             inverse);
+            kokkos_gntr_sv_r.applyGenerator("RX", {control_wire}, {true},
+                                            {wire}, inverse);
+
+            auto result_gntr_sv_cr = kokkos_gntr_sv_cr.getDataVector();
+            auto result_gntr_sv_r = kokkos_gntr_sv_r.getDataVector();
+
+            for (std::size_t j = 0; j < exp2(num_qubits); j++) {
+                CHECK(imag(result_gntr_sv_cr[j]) ==
+                      Approx(imag(result_gntr_sv_r[j])).margin(ep_margin));
+                CHECK(real(result_gntr_sv_cr[j]) ==
+                      Approx(real(result_gntr_sv_r[j])).margin(ep_margin));
+            }
+        }
+    }
+
+    SECTION("CRY") {
+        StateVectorKokkos<TestType> kokkos_gntr_sv_cr{ini_st.data(),
+                                                      ini_st.size()};
+        StateVectorKokkos<TestType> kokkos_gntr_sv_r{ini_st.data(),
+                                                     ini_st.size()};
+
+        if (control_wire == wire) {
+            PL_REQUIRE_THROWS_MATCHES(
+                kokkos_gntr_sv_r.applyGenerator("RX", {control_wire}, {true},
+                                                {wire}, inverse),
+                LightningException,
+                "`controlled_wires` and `target wires` must be disjoint.");
+        } else {
+            kokkos_gntr_sv_cr.applyGenerator("CRY", {control_wire, wire},
+                                             inverse);
+            kokkos_gntr_sv_r.applyGenerator("RY", {control_wire}, {true},
+                                            {wire}, inverse);
+
+            auto result_gntr_sv_cr = kokkos_gntr_sv_cr.getDataVector();
+            auto result_gntr_sv_r = kokkos_gntr_sv_r.getDataVector();
+
+            for (std::size_t j = 0; j < exp2(num_qubits); j++) {
+                CHECK(imag(result_gntr_sv_cr[j]) ==
+                      Approx(imag(result_gntr_sv_r[j])).margin(ep_margin));
+                CHECK(real(result_gntr_sv_cr[j]) ==
+                      Approx(real(result_gntr_sv_r[j])).margin(ep_margin));
+            }
```

Comment:
Can we use our Approx function to handle complex numbers?

Reply:
Indeed, I can update it to 
```
REQUIRE(isApproxEqual(kokkos_gntr_sv_cr.getData(), kokkos_gntr_sv_cr.getLength(), kokkos_gntr_sv_r.getData(), kokkos_gntr_sv_r.getLength(), ep_margin));
```
, but I think I will wait until [Ali's PR](https://github.com/PennyLaneAI/pennylane-lightning/pull/1000) to be merged, which updates the `isApproxEqual` function to
```
REQUIRE(isApproxEqual(kokkos_gntr_sv_cr.getDataVector(),
                                  kokkos_gntr_sv_r.getDataVector()));
```

After the merge I can update all the tests to use this.

Reply:
Sounds good!

Reply:
Feel free to merge as is, and add a story in shortcut to update it later.
We may have a similar situation in other places.